services/motion/builtin/builtin.go

```diff
@@ -385,30 +483,43 @@ func (ms *builtIn) planMoveOnGlobe(
 		return nil, nil, err
 	}
 
-	// make call to motionplan
-	solutionMap, err := motionplan.PlanMotion(
-		ctx,
-		ms.logger,
-		referenceframe.NewPoseInFrame(referenceframe.World, goal),
-		offsetFrame,
-		inputMap,
-		fs,
-		wrldst,
-		nil,
-		extra,
-	)
-	if err != nil {
-		return nil, nil, err
-	}
+	return &motionplan.PlanRequest{
+		Logger:      ms.logger,
+		Goal:        referenceframe.NewPoseInFrame(referenceframe.World, goal),
+		Frame:       offsetFrame,
+		FrameSystem: fs,
+		Inputs:      referenceframe.StartPositions(fs),
+		WorldState:  worldState,
+		Options:     extra,
+	}, kb, nil
+}
 
-	plan, err := motionplan.FrameStepsFromRobotPath(kbf.Name(), solutionMap)
+func (ms *builtIn) executePlan(ctx context.Context, kinematicBase kinematicbase.KinematicBase, plan motionplan.Plan) error {
+	waypoints, err := plan.GetFrameSteps(kinematicBase.Name().Name)
 	if err != nil {
-		return nil, nil, err
+		return err
 	}
-	for _, step := range plan {
-		ms.logger.Info(step)
+
+	for i := 1; i < len(waypoints); i++ {
+		select {
+		case <-ctx.Done():
+			return nil
+		default:
+			ms.logger.Info(waypoints[i])
+			segment := []referenceframe.Input{
```

Comment:
This utterly breaks TP-space.

Also, I'm pretty sure the inputs for a diff drive kinematic base are relative to the location at the start of the move, not relative to its current origin, so isn't this unnecessary for a diff drive base as well?

Reply:
Fixed this